oat-sa/extension-tao-testqti
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1418](https://oat-sa.atlassian.net/browse/ADF-1418)

Requires:
 - [x] https://github.com/oat-sa/tao-test-runner-fe/pull/64
 - [x] https://github.com/oat-sa/tao-test-runner-qti-fe/pull/513
 - [x] update the file [`package.json`](https://github.com/oat-sa/extension-tao-testqti/pull/2375/files#diff-03c6d048c4955944172f34af2a152ae6cbd4ffc2ec11fcd4bc7cd44b194f249cR11-R12) once the companion PRs have been merged and released

### Summary

Update the dependencies in order to support further versions of Node

BREAKING CHANGE:
- The package now requires Node.js 14.17+
- Replace Node-Sass with Dart-Sass

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

### How to test
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package:
    ```
    cd taoQtiTest/views
    npm i
    cd ../../tao/views
    cd build
    npm i
    npx grunt taoqtitestsass
    npx grunt taoqtitestbundle
    npx grunt connect:dev taoqtitesttest
    ```

**Note** After switching to Node 18, you may need to erase the node_module folder, otherwise the error `>> Error: Could not locate the bindings file` may arise.


[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ADF-1418]: https://oat-sa.atlassian.net/browse/ADF-1418?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
